octoprint_printhistory/parser.py

```diff
@@ -0,0 +1,294 @@
+            elif line == "\n":
+                continue
+            else:
+                break
+        file.seek(0)
+        return parameters
+
+    def parse_bottom(self, file):
+        parameters = {}
+        for line in reverse_readline(file):
+            if line.startswith(";"):
+                line = line.split(";")[1].strip()
+                if self.is_parameter.match(line):
+                    splitted = line.split("=")
+                    param = splitted[0].strip()
+                    value = splitted[1].strip()
+                    parameters.update({param: value})
+            elif line == "\n":
+                continue
+            else:
+                break
+        return parameters
+
+
+class Simplify3DParser(BaseParser):
+
+    def __init__(self):
+        self.name = "simplify3d"
+        self.version = None
+        # checks if string starts with a word followed by a comma
+        self.is_parameter = re.compile(r"^\w+,{1}")
+        self.is_bparameter = re.compile(r"^[\w ]+:{1}")
+
+    def detect(self, file):
+        detected = False
+        # on the first line
+        # ; G-Code generated by Simplify3D(R) Version 3.1.0
+        line = file.readline()
+        file.seek(0)
+        if re.search(r"Simplify3D\(R\)", line):
+            self.version = VERSION_REGEX.search(line).group(0)
+            detected = True
+        return detected
+
+    def parse_header(self, file):
+        parameters = {}
+        for line in file:
+            if line.startswith(";"):
+                # extract a string between semicolons ; =>target string<= ;
+                line = line.split(";")[1].strip()
+                if self.is_parameter.match(line):
+                    comma_split = line.split(",")
+                    param = comma_split[0]
+                    value = ",".join(comma_split[1:])
+                    try:
+                        parameters.update({param: value})
+                    except ValueError:
+                        pass
+            # comment section is ended
+            else:
+                break
+        file.seek(0)
+        return parameters
+
+    def parse_bottom(self, file):
+        # Filament length is a duplicate
+        # ; Build Summary
+        # ;   Build time: 3 hours 30 minutes
+        # ;   Filament length: 54599.6 mm (54.60 m)
+        # ;   Plastic volume: 131327.49 mm^3 (131.33 cc)
+        # ;   Plastic weight: 164.16 g (0.36 lb)
+        # ;   Material cost: 20.52
+        parameters = {}
+        for line in reverse_readline(file):
+            if line.startswith(";"):
+                line = line.split(";")[1].strip()
+                if self.is_bparameter.match(line):
+                    splitted = line.split(":")
+                    param = splitted[0]
+                    value = splitted[1].strip()
+                    parameters.update({param: value})
+            else:
+                break
+        # Rename "Filament length"
```

Comment:
ah, I get it. Then do just

```python
parameters['Filament length Gcode'] = parameters.pop('Filament length')
```

source: http://stackoverflow.com/a/16475444/699864

And remove that useless comment. (It's useless because you shouldn't explain simple oneliner.)

And THEN you can drop the whole try-except statement since you'll no longer need it.